tools/commonality_finder/CandD.xml

```diff
@@ -0,0 +1,43 @@
+<tool id="commonality_r" name="commonality_finder" version="1.1.0">
+    <description>finds commonality</description>
+    <requirements>
+       <requirement type="package" version="3.6">r-base</requirement>
+    </requirements>
+    <command><![CDATA[
+		ln -s '$input1' input1.csv &&
+		ln -s '$input2' input2.csv &&
+		ln -s '$input3' input3.csv &&
+		ln -s '$input4' input4.csv &&
+		ln -s '$input5' input5.csv &&
+		ln -s '$input6' input6.csv &&
+		
+		Rscript '$__tool_directory__/commonality-working feb 21 20.R'
+    ]]></command>
+    <inputs>
+        <param format="csv" name="input1" type="data" label="First Substrate Set"/>
+        <param format="csv" name="input2" type="data" label="First Substrate Background Frequency"/>
+        <param format="csv" name="input3" type="data" label="Second Substrate Set"/>
+        <param format="csv" name="input4" type="data" label="Second Substrate Background Frequency"/>
+        <param format="csv" name="input5" type="data" label="Third Substrate Set"/>
+        <param format="csv" name="input6" type="data" label="Third Substrate Background Frequency"/>
+        <param name="outGroup" type="text" value="kinase" label="Kinase Name"/>
+    </inputs>      
+    <outputs>
+        <data format="csv" name="substrates" from_work_dir="sharedmotifs.csv" label="${outGroup}_shared Substrates.csv"/>
+        <data format="csv" name="SBF" from_work_dir="sharedSBF.csv" label="${outGroup}_shared SBF.csv"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="input1.csv" ftype="csv" value="input1.csv"/>
+            <param name="positives" ftype="csv" value="substrates.csv"/>
+            <param name="SBF" ftype="csv" value="SBF.csv"/>
+			<output name="Negatives" file="negatives.csv"/>

```

Comment:
The output name in the test has to one of the outputs you declared in the tool, in this case you have either "substrates" or "SBF" as the name of an output.

Reply:
OK, I've worked to fix all the tests file names